### ISSUE TITLE
Rename root module to `Meilisearch` insteaf of `MeiliSearch`

### ISSUE THREAD

**Issue by ellnix** [meilisearch/meilisearch-rails#347]
> Also, if you have time you could put in your big breaking changing the main class name to `Meilisearch` instead of `MeiliSearch`. Not only here but in the ruby gem as well.

_Originally posted by @brunoocasali in https://github.com/meilisearch/meilisearch-rails/pull/344#discussion_r1507561897_

**Comment by jason-hobbs** [comment]
I am starting to get the deprecation warning: WARN -- : [meilisearch-ruby] The top-level module of Meilisearch has been renamed.
[meilisearch-ruby] Please update "MeiliSearch" to "Meilisearch".

**Comment by ellnix** [comment]
> I am starting to get the deprecation warning: WARN -- : [meilisearch-ruby] The top-level module of Meilisearch has been renamed. [meilisearch-ruby] Please update "MeiliSearch" to "Meilisearch".

Do you happen to have `meilisearch-ruby` in your Gemfile? It shouldn't normally pull 0.30.

**Comment by ellnix** [comment]
@jason-hobbs I fixed the version constraint, this warning should not appear if you use the just-released 0.14.2.

Please let me know if it worked.

**Comment by ansonhoyt** [comment]
Worked for me, though `bundle update meilisearch-rails` wouldn't upgrade from `0.14.1` to `0.14.2` until I manually cleared this from my `Gemfile.lock`:

```
    meilisearch (0.30.0)
      httparty (~> 0.17)
```

**Comment by ellnix** [comment]
> Worked for me, though `bundle update meilisearch-rails` wouldn't upgrade from `0.14.1` to `0.14.2` until I manually cleared this from my `Gemfile.lock`:
> 
> ```
>     meilisearch (0.30.0)
>       httparty (~> 0.17)
> ```

Seems understandable, it's a very strange situation we are in. I can only hope that anyone on 0.14.1 has meilisearch-ruby locked already, and ones who do not will get 0.14.2... 🫤

**Comment by jason-hobbs** [comment]
Yes that worked for me.  I locked the version, did a bundle install and it stated:
Fetching meilisearch 0.28.4 (was 0.30.0)
Installing meilisearch 0.28.4 (was 0.30.0)
Fetching meilisearch-rails 0.14.2 (was 0.14.1)
Installing meilisearch-rails 0.14.2 (was 0.14.1)
Bundle complete! 66 Gemfile dependencies, 190 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.


Warning went away.  Thank you!
